### PR TITLE
Add branch and release guidance in contributing.md

### PR DESCRIPTION
**Pull request by polita**
@tushargupta51 @brentschmaltz @bgever
I added a CONTRIBUTING.md with branch, build, test, and contribution information. I added a link to it in the readme.md. This will go into master so it's on the repo landing page. This addresses #402.

When we create new branches, we'll need to make sure to keep this up to date.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# Branch Structure
+* **master**: The latest official GA version, which is currently 4.x
+* **dev**: The dev working branch of master, which is 4.x
+* **KMaster**: The master branch for 5.x (Project K)
```
Comment (tushargupta51) [inline CONTRIBUTING.md]: Do we want to use the term "Project K"?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+* **KDev**: The dev working branch of KMaster (Project K)
+
+If you are contributing code to 4.x, you should branch from **dev**
+If you are contributing code to 5.x (Project K), you should branch from **KDev**
+
+## Release branches
+Release tracking branches are created for each significant release. At this time, we have the following release branches:
+* **KDevRC2**
```
Comment (tushargupta51) [inline CONTRIBUTING.md]: KDevRc2 is a topic branch.

File: CONTRIBUTING.md (hunk 1 above)
Comment (polita) [inline CONTRIBUTING.md]: Good question. What are they calling it?

File: README.md [hunk 3]
```diff
@@ -15,22 +15,45 @@ We recommend you use the "adal" tag so we can see it! Here is the latest Q&A on
...
+
+#### Microsoft.IdentityModel.Protocols.WsFederation (version 2.x, currently in beta)
+* Includes types that provide support for WsFederation protocol.
+
+#### Microsoft.IdentityModel.Logging (version 1.x)
+* Includes Event Source based logging support.
+
 ## License
 
 Copyright (c) Microsoft Open Technologies, Inc.  All rights reserved. Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (brentschmaltz) [inline README.md]: MOT is no longer referenced.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+## Legal
+You will need to complete a Contributor License Agreement (CLA). Briefly, this agreement testifies that you are granting us permission to use the submitted change according to the terms of the project's license, and that the work being submitted is under appropriate copyright.
+
+Please submit a Contributor License Agreement (CLA) before submitting a pull request. You may visit https://cla.microsoft.com to sign digitally. Once we have received the signed CLA, we'll review the request. 
+
+## Housekeeping
+Your pull request should: 
+
+* Include a description of what your change intends to do
+* Be a child commit of a reasonably recent commit in the correct branch
```
Comment (brentschmaltz) [inline CONTRIBUTING.md]: word smith "child commit"

**Comment by polita** [comment]
Alright, I fixed the 3.x assembly list. @tushargupta51 and @brentschmaltz, can you take a look?

**Comment by tushargupta51** [comment]
@polita You might need to rebase against master again since @brandwe recently updated readme with security reporting section. :shipit:

**Pull request merged**